examples/meta/src/gaussian_processes/gaussian_process_regression.sg

```diff
@@ -0,0 +1,47 @@
+CSVFile f_feats_train("../../data/regression_1d_sinc_features_train.dat")
+CSVFile f_feats_test("../../data/regression_1d_sinc_features_test.dat")
+CSVFile f_labels_train("../../data/regression_1d_sinc_labels_train.dat")
+CSVFile f_labels_test("../../data/regression_1d_sinc_labels_test.dat")
+
+#![create_features]
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+RegressionLabels labels_train(f_labels_train)
+RegressionLabels labels_test(f_labels_test)
+#![create_features]
+
+#![create_appropriate_kernel_and_mean_function]
+Real width = 1
+GaussianKernel kernel(features_train, features_train, width)
+ZeroMean mean_function()
+#![create_appropriate_kernel_and_mean_function]
+
+#![create_instance]
+GaussianLikelihood gauss_likelihood()
+ExactInferenceMethod inference_method(kernel, features_train, mean_function, labels_train, gauss_likelihood)
+GaussianProcessRegression gp_regression(inference_method)
+#![create_instance]
+
+#![train_and_apply]
+gp_regression.train()
+RegressionLabels labels_predict = gp_regression.apply_regression(features_test)
+#![train_and_apply]
+
+#![optimize_marginal_likelihood]
+GradientCriterion grad_criterion()
```

Comment:
@yorkerlin I am mostly thinking about a simple convenience method that does all the above calls, and hides them behind a single `optimize` call, which always optimizes all variables via ML2. I think this should already be done in order to minimize obscure code.